src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -150,56 +176,100 @@ loadMesh( Path const & filePath )
+          hexa->GetPointIds()->SetId( 0, i+j*nx+k*ny*nx );
+          hexa->GetPointIds()->SetId( 1, i+1+j*nx+k*ny*nx );
+          hexa->GetPointIds()->SetId( 2, i+1+(j+1)*nx+k*ny*nx );
+          hexa->GetPointIds()->SetId( 3, i+(j+1)*nx+k*ny*nx );
+          hexa->GetPointIds()->SetId( 4, i+j*nx+(k+1)*ny*nx );
+          hexa->GetPointIds()->SetId( 5, i+1+j*nx+(k+1)*ny*nx );
+          hexa->GetPointIds()->SetId( 6, i+1+(j+1)*nx+(k+1)*ny*nx );
+          hexa->GetPointIds()->SetId( 7, i+(j+1)*nx+(k+1)*ny*nx );
+          cells->InsertNextCell( hexa );
+        }
+      }
+    }
+  }
+  return cells;
+}
+
 template< typename INDEX_TYPE, typename POLICY >
 ArrayOfArrays< INDEX_TYPE, INDEX_TYPE >
-buildElemToNodesImpl( vtkUnstructuredGrid & mesh )
+buildElemToNodesImpl( vtkDataSet & mesh )
 {
   localIndex const numCells = LvArray::integerConversion< localIndex >( mesh.GetNumberOfCells() );
   array1d< INDEX_TYPE > nodeCounts( numCells );
-  vtkCellArray & cells = *mesh.GetCells();
+  const vtkNew< vtkCellArray > & cells = GetCellArray( mesh ); // changed *mesh.GetCells(); to GetCellArray(mesh)
 
   // GetCellSize() is always thread-safe, can run in parallel
   forAll< parallelHostPolicy >( numCells, [nodeCounts = nodeCounts.toView(), &cells] ( localIndex const cellIdx )
   {
-    nodeCounts[cellIdx] = LvArray::integerConversion< INDEX_TYPE >( cells.GetCellSize( cellIdx ) );
+    nodeCounts[cellIdx] = LvArray::integerConversion< INDEX_TYPE >( cells->GetCellSize( cellIdx ) );
   } );
 
   ArrayOfArrays< INDEX_TYPE, INDEX_TYPE > elemToNodes;
   elemToNodes.template resizeFromCapacities< parallelHostPolicy >( numCells, nodeCounts.data() );
-
   vtkIdTypeArray const & globalPointId = *vtkIdTypeArray::FastDownCast( mesh.GetPointData()->GetGlobalIds() );
 
   // GetCellAtId() is conditionally thread-safe, use POLICY argument
   forAll< POLICY >( numCells, [&cells, &globalPointId, elemToNodes = elemToNodes.toView()] ( localIndex const cellIdx )
   {
     vtkIdType numPts;
     vtkIdType const * points;
-    cells.GetCellAtId( cellIdx, numPts, points );
+    cells->GetCellAtId( cellIdx, numPts, points );
     for( int a = 0; a < numPts; ++a )
     {
       vtkIdType const pointIdx = globalPointId.GetValue( points[a] );
       elemToNodes.emplaceBack( cellIdx, LvArray::integerConversion< INDEX_TYPE >( pointIdx ) );
     }
   } );
-
   return elemToNodes;
 }
 
 template< typename INDEX_TYPE >
 ArrayOfArrays< INDEX_TYPE, INDEX_TYPE >
-buildElemToNodes( vtkUnstructuredGrid & mesh )
+buildElemToNodes( vtkDataSet & mesh )
 {
   // According to VTK docs, IsStorageShareable() indicates whether pointers extracted via
   // vtkCellArray::GetCellAtId() are pointers into internal storage rather than temp buffer
   // and thus results can be used in a thread-safe way.
-  return mesh.GetCells()->IsStorageShareable()
+  return GetCellArray( mesh )->IsStorageShareable()                        // changed mesh.GetCells() to GetCellArray
```

Comment:
Careful, here you are building again the list of cells, this can be costly. I suggest creating the list here (`cells= GetCellArray( mesh )`), and then passing `cells` to `buildElemToNodesImpl` as an argument. This requires changing the signature of the method.